service_api.md

```diff
@@ -0,0 +1,284 @@
+
+    class ProjectCreationRequest {
+        String projectId
+        ProjectInformation requestBody
+        String requestId
+    }
+
+    class ProjectCreationResponse {
+        String projectId
+        ProjectInformation responseBody
+        String requestId
+    }
+
+    class ProjectInformation
+
+    ProjectCreationRequest ..> ProjectInformation
+    ProjectCreationResponse ..> ProjectInformation
+```
+
+### Creating a sub-resource
+Creating a sub-resource that are in a many-to-one relationship with their parent resource should be made easy.
+
+For example, creating an `Example Group` for an existing `Experiment` shall be a no-brainer. Therefore, it is 
+adviced that the API provides semantically matching request objects for these tasks, that contain
+the required identifiers to enable the service to create these resource enrichments.
+
+For example, the method call for creating a new `Example Group` could be `create(ExperimentGroupCreationRequest request)` and the request
+contains the required IDs to enable the service to perform the task.
+
+```mermaid
+---
+title: Create Experiment Group Example
+---
+classDiagram
+    class AsyncProjectService {
+        Mono~ExperimentGroupCreationResponse~ create(ExperimentGroupCreationRequest request)
+    }
+    <<interface>> AsyncProjectService
+    AsyncProjectService ..> ExperimentGroupCreationRequest
+    AsyncProjectService ..> ExperimentGroupCreationResponse
+
+    class ExperimentGroupCreationRequest {
+        String projectId
+        String experimentId
+        ExperimentalGroup group
+        String requestId
+    }
+
+    class ExperimentGroupCreationResponse {
+        String projectId
+        String experimentId
+        ExperimentalGroup group
+        String requestId
+    }
+
+    class ExperimentalGroup {
+        String id
+        ...
+    }
+
+    ExperimentGroupCreationRequest ..> ExperimentalGroup
+    ExperimentGroupCreationResponse ..> ExperimentalGroup
+```
+
+## Get examples
+
+### Get a specific project
+
+```mermaid
+---
+title: Get Project Example
+---
+classDiagram
+    class AsyncProjectService {
+        Mono~ProjectInformation~ getProject(String id)
+    }
+    <<interface>> AsyncProjectService
+```
+
+### Get all projects
+
+```mermaid
+---
+title: Get Projects Example
+---
+classDiagram
+    class AsyncProjectService {
+        Flux~ProjectInformation~ getProjects()
+    }
+    <<interface>> AsyncProjectService
+```
+
+### Get experiment
+
+```mermaid
+---
+title: Get Experiment Example
+---
+classDiagram
+    class AsyncProjectService {
+        Mono~ExperimentInformation~ getExperiment(String projectId, String experimentId)
+    }
+    <<interface>> AsyncProjectService
+```
+
+### Get all experiments for a project
+
+```mermaid
+---
+title: Get Experiments Example
+---
+classDiagram
+    class AsyncProjectService {
+        Flux~ExperimentInformation~ getExperiment(String projectId, String experimentId)
```

Comment:
This will not compile as the method name and parameters are the same and thus java cannot decide which method to use. This method should be named `getExperiments` with an `s`.